### PR TITLE
Improve godocs. Add sample code for func(Amount)Format

### PR DESCRIPTION
**Pull request by wallclockbuilder**
Augment the multi-line godoc comment with a clear concise code
sample.

### CODE REVIEW

**Comment by wallclockbuilder** [comment]
Before
1 format multi-line comment

**Comment by wallclockbuilder** [comment]
After
2 format multi-line augmented with sample code

File: example_test.go [hunk 1]
```diff
@@ -74,3 +74,17 @@ func ExampleAmount_unitConversions() {
 	// Satoshi to MicroBTC: 444333222111 μBTC
 	// Satoshi to Satoshi: 44433322211100 Satoshi
 }
+func ExampleAmount_Format() {
```
Comment (davecgh) [inline example_test.go]: Nitpick: Please add a blank line above the function definition for consistency.

**Comment by wallclockbuilder** [comment]
Added indentation.
Squashed commits.

**Pull request closed without merging**